ui/de-lib/src/main/java/org/iplantc/de/apps/client/views/submit/SubmitAppForPublicUseViewImpl.java

```diff
@@ -265,6 +267,21 @@ public String getPath() {
         tree.setCheckable(true);
         tree.setCheckStyle(CheckCascade.CHILDREN);
         tree.setCheckNodes(CheckNodes.LEAF);
+        tree.getSelectionModel().addSelectionHandler(new SelectionHandler<AppCategory>() {
+            @Override
+            public void onSelection(SelectionEvent<AppCategory> event) {
+                AppCategory selectedItem = event.getSelectedItem();
+                // Have to deselect after un/checking the item, or else a second
+                // SelectionEvent gets fired and reverts the check
+                if (tree.isChecked(selectedItem)) {
+                    tree.setChecked(selectedItem, Tree.CheckState.UNCHECKED);
+                    tree.getSelectionModel().deselect(selectedItem);
+                } else {
+                    tree.setChecked(selectedItem, Tree.CheckState.CHECKED);
+                    tree.getSelectionModel().deselect(selectedItem);
```

Comment:
It looks like the `SelectionModel` should be selecting the `selectedItem`, since it's also getting `CHECKED` here.